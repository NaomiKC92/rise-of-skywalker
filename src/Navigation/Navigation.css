* {
  margin: 0;
  padding: 0;
}

.navigation-pane {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.navigation-button {
  background: #ABABAB;
  border-radius: 10px;
  border: none;
  color: #000;
  font-size: 1rem;
  font-weight: bold;
  margin-right: 0.5rem;
  margin: 10px;
  outline: none;
  padding: 0.4rem 1.3rem;
  transition: opacity 0.2s ease-in;
  width: 200px;
}

.navigation-button:hover {
  cursor: pointer;
}

.nav-card {
  align-items: center;
  background-color: #000;
  display: flex;
  justify-content: space-between;
  margin-bottom: 1rem;
  min-height: 200px;
  opacity: 0.9;
  padding: 0.7rem 2rem;
  width: 100%;
  z-index: 1;
}

.user-name,
.user-quote,
.user-rank {
  color: #FFF;
  font-size: 25px;
  font-weight: bold;
}
