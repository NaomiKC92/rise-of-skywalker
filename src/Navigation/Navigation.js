import React from 'react';
import { Link } from 'react-router-dom';
import './Navigation.css';

const Navigation = ({
  userName,
  userQuote,
  userRank,
  stateUpdateIsCharactersDataLoaded,
  numberOfFavoriteCharacters,
  clearFavoritesUponSignout
}) => {
  return (
    <div className='card'>
      <div className='user-profile'>
        <p>{userName}</p>
        <p>"{userQuote}"</p>
        <p>{userRank}</p>
      </div>
      <div className='navigation-pane'>
        <Link to='/favorites'>
          <button className='navigation-button'>
            Favorite Characters ({numberOfFavoriteCharacters.length})
          </button>
        </Link>
        <Link to='/movies'>
          <button
            onClick={stateUpdateIsCharactersDataLoaded}
            className='navigation-button'>
            Back to All Episodes
          </button>
        </Link>
        <Link to='/'>
          <button
            onClick={
              (stateUpdateIsCharactersDataLoaded, clearFavoritesUponSignout)
            }
            className='navigation-button'>
            Sign Out
          </button>
        </Link>
      </div>
    </div>
  );
};

export default Navigation;
