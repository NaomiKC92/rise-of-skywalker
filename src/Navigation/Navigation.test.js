import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './Navigation';

describe('Navigation', () => {
  let wrapper;

  const mockUserName = 'Peerat';
  const mockUserQuote = 'Naomi is amazing!';
  const mockUserRank = 'Jedi Master';
  const mockStateUpdateIsCharactersDataLoaded = jest.fn();
  const mockNumberOfFavoriteCharacters = 5;
  const mockClearFavoritesUponSignout = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Navigation
        userName={mockUserName}
        userQuote={mockUserQuote}
        userRank={mockUserRank}
        stateUpdateIsCharactersDataLoaded={
          mockStateUpdateIsCharactersDataLoaded
        }
        numberOfFavoriteCharacters={mockNumberOfFavoriteCharacters}
        clearFavoritesUponSignout={mockClearFavoritesUponSignout}
      />
    );
  });

  it('should match the Navigation Component Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
