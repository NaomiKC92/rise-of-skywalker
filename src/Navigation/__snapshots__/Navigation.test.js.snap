// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match the Navigation Component Snapshot 1`] = `
<div
  className="card"
>
  <div
    className="user-profile"
  >
    <p>
      Peerat
    </p>
    <p>
      "
      Naomi is amazing!
      "
    </p>
    <p>
      Jedi Master
    </p>
  </div>
  <div
    className="navigation-pane"
  >
    <Link
      to="/favorites"
    >
      <button
        className="navigation-button"
      >
        Favorite Characters (
        )
      </button>
    </Link>
    <Link
      to="/movies"
    >
      <button
        className="navigation-button"
        onClick={[MockFunction]}
      >
        Back to All Episodes
      </button>
    </Link>
    <Link
      to="/"
    >
      <button
        className="navigation-button"
        onClick={[MockFunction]}
      >
        Sign Out
      </button>
    </Link>
  </div>
</div>
`;
